src/Utility/Hash.php

```
@@ -161,7 +161,9 @@ public static function extract(array $data, $path)
             if ($conditions) {
                 $filter = [];
                 foreach ($next as $item) {
-                    if (is_array($item) && static::_matches($item, $conditions)) {
+                    if ((is_array($item) || $item instanceof ArrayAccess) &&
+                        static::_matches($item, $conditions)
```

Should we raise an exception if `$data` is not an array or ArrayAccess?